### PR TITLE
Add protocol to Metric Configuration Service

### PR DESCRIPTION
**Pull request by wtong98**
### Changes
* New protocol for metric configuration service
* This PR is an updated version of #155

### Goal
We introduce a new protocol for dynamically configuring metric collection schedules. It follows from this [OTEP](https://github.com/open-telemetry/oteps/pull/121).

An example implementation is almost completed and will be sent to the collector-contrib and go-contrib repos. In the meantime, we'd love to hear your thoughts on this protocol and the [related specification](https://github.com/open-telemetry/opentelemetry-specification/pull/751)!

### About
We're an intern group at Google.  Members include:
* Ming Chen @vmingchen (host)
* Josh Montana @jdmontana (co-host)
* Churan Chen @churanchen (intern, now done)
* William Tong @wtong98 (me! also an intern)

### CODE REVIEW

**Comment by wtong98** [comment]
@bogdandrutu Great thanks! The local setup was a little weird, since we based our changes off the 0.4.0 release to avoid dependency problems in our prototypes. Sorry I missed your earlier message! I went ahead and straightened out the history for this PR and force-pushed the changes.

File: opentelemetry/proto/collector/README.md [hunk 1]
```diff
@@ -4,6 +4,5 @@ This package describes the OpenTelemetry collector protocol.
 
 ## Packages
 
-1. `common` package contains the common messages shared between different services.
```
Comment (tigrannajaryan) [inline opentelemetry/proto/collector/README.md:7]: Why is `common` removed? It is still a package, right?

Comment (wtong98) [reply opentelemetry/proto/collector/README.md]: Oops! No idea how that happened. Changes to that file have been reverted.

File: opentelemetry/proto/experimental/metricconfigservice/metricconfig_service.proto [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+
+import "opentelemetry/proto/resource/v1/resource.proto";
+
+option java_multiple_files = true;
+option java_package = "io.opentelemetry.proto.experimental.metricconfigservice";
+option java_outer_classname = "MetricConfigServiceProto";
+option go_package = "github.com/open-telemetry/opentelemetry-proto/gen/go/experimental/metricconfigservice";
+
+// MetricConfig is a service that enables updating metric schedules, trace
+// parameters, and other configurations on the SDK without having to restart the
```
Comment (anuraaga) [inline opentelemetry/proto/experimental/metricconfigservice/metricconfig_service.proto:27]:
```suggestion
// MetricConfig is a service that enables updating metric schedules and other configurations on the SDK without having to restart the
```

Looks like the scope was narrowed to just metrics which makes sense

File: opentelemetry/proto/experimental/metricconfigservice/metricconfig_service.proto [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+
+package opentelemetry.proto.experimental.metricconfigservice;
```
Comment (bogdandrutu) [inline opentelemetry/proto/experimental/metricconfigservice/metricconfig_service.proto:17]: add this in:
"experimental/metrics/configservice"

Comment (wtong98) [reply opentelemetry/proto/experimental/metricconfigservice/metricconfig_service.proto]: Done

**Comment by SergeyKanzhelev** [comment]
Thank you for initiating the experimental folder!

**Pull request merged**